lib/slide_hub/cloud/engine/aws.rb

```diff
@@ -0,0 +1,280 @@
+            receipt_handle: message_object.handle,
+          },)
+          resp
+        end
+
+        def self.batch_delete(entries)
+          # self.sqs.delete_message_batch(queue_url: @config.sqs_url, entries: entries)
+          result = true
+          entries.each do |entry|
+            result = self.delete_message(entry)
+          end
+          result
+        end
+
+        ## S3
+        #
+        def self.upload_files(bucket, files, prefix)
+          files.each do |f|
+            Aws::S3::Client.new(region: @config.region).put_object(
+              bucket: bucket,
+              key: "#{prefix}/#{File.basename(f)}",
+              body: File.read(f),
+                acl: 'public-read',
+                storage_class: 'REDUCED_REDUNDANCY',
+            ) if File.exist?(f)
+          end
+        end
+
+        def self.get_file_list(bucket, prefix)
+          resp = Aws::S3::Client.new(region: @config.region).list_objects({
+            bucket: bucket,
+            max_keys: 1000,
+            prefix: prefix,
+          },)
+          files = []
+          resp.contents.each do |f|
+            files.push({ key: f.key })
+          end
+          files
+        end
+
+        def self.save_file(bucket, key, destination)
+          Aws::S3::Client.new(region: @config.region).get_object(
+            response_target: destination,
+            bucket: bucket,
+            key: key)
+        end
+
+        def self.delete_slide(key)
+          if key.empty?
+            return false
+          end
+          files = self.get_file_list(@config.bucket_name, key)
+          self.delete_files(@config.bucket_name, files)
+          true
+        end
+
+        def self.delete_generated_files(key)
+          if key.empty?
+            return false
+          end
+          files = self.get_file_list(@config.image_bucket_name, key)
+          self.delete_files(@config.image_bucket_name, files)
+          true
+        end
+
+        def self.get_slide_download_url(key)
+          self.get_download_url(@config.bucket_name, key)
+        end
+
+        def self.delete_files(bucket, files)
+          Aws::S3::Client.new(region: @config.region).delete_objects({
+            bucket: bucket,
+            delete: {
+              objects: files,
+              quiet: true,
+            },
+          },) unless files.empty?
+        end
+
+        def self.get_download_url(bucket, key)
+          signer = Aws::S3::Presigner.new(client: Aws::S3::Client.new(region: @config.region))
+          url = signer.presigned_url(:get_object, bucket: bucket, key: key)
+          url
+        end
+
+        def self.create_policy
+          base_time = Time.zone.now.in_time_zone('UTC')
+
+          if !@config.aws_access_id.blank? && !@config.aws_secret_key.blank?
+            access_id = @config.aws_access_id
+            secret_key = @config.aws_secret_key
+            security_token = ''
+          else
+            ec2 = Aws::EC2::Client.new
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'ec2' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)